dev/reference/sql/statements/split-region.md

```diff
@@ -0,0 +1,209 @@
+
+在 TiDB 中新建一个表后，默认会单独切分出 1 个 region 来存储这个表的数据，这个默认行为由配置文件中的 `split-table` 控制。当这个 region 中的数据超过默认 region 大小限制后，这个 region 会开始分裂成 2 个 region。
+
+上述情况中，如果在新建的表上发生大批量写入，则会造成热点，因为开始只有一个 region，所有的写请求都发生在该 region 所在的那台 TiKV 上。
+
+为解决上述场景中的热点问题，TiDB 引入了预切分 region 的功能，即可以根据用户指定的参数，预先为某个表切分出多个 region，并打散到各个 TiKV 上去。
+
+## Split region 的使用
+
+Split region 有 2 种不同的语法，具体如下：
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE table_name [INDEX index_name] BETWEEN (lower_value) AND (upper_value) REGIONS region_num
+```
+
+`BETWEEN lower_value AND upper_value REGIONS region_num` 语法是通过指定上、下边界和 region 数量，然后在上、下边界之间均匀切分出 `region_num` 个 region。
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE table_name [INDEX index_name] BY (value_list) [, (value_list)] ...
+```
+
+`BY value_list…` 语法将手动指定一系列的点，然后根据这些指定的点切分 region，适用于数据不均匀分布的场景。
+
+### Split Table region
+
+表中行数据的 key 由 `table_id` 和 `row_id` 编码组成，格式如下：
+
+```go
+t[table_id]_r[row_id]
+```
+
+例如，当 `table_id` 是 22，`row_id` 是 11 时：
+
+```go
+t22_r11
+```
+
+同一表中行数据的 `table_id` 是一样的，但 `row_id` 肯定不一样，所以可以根据 `row_id` 来切分 region。
+
+#### 均匀切分
+
+由于 `row_id` 是整数，所以根据用户指定的 `lower_value`、`upper_value` 以及 `region_num`，可以很容易地计算出需要切分的 key。先计算出 step（`step = (upper_value - lower)/num`），然后在 `lower` 和 `upper` 之间每隔 step 区间切一次，最终切出 `num` 个 region。
+
+例如，对于表 t，如果想要从 `minInt64`~`maxInt64` 之间均匀切割出 16 个 region，可以用以下语句：
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE t BETWEEN (-9223372036854775808) AND (9223372036854775807) REGIONS 16;
+```
+
+该语句会把表 t 从 minInt64 到 maxInt64 之间均匀切割出 16 个 region。如果已知主键的范围没有这么大，比如只会在 0~1000000000 之间，那可以用 0 和 1000000000 分别代替上面的 minInt64 和 maxInt64 来切分 region。
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE t BETWEEN (0) AND (1000000000) REGIONS 16;
+```
+
+#### 不均匀切分
+
+如果已知数据不是均匀分布的，比如想要 -inf ~ 10000 切一个 region，10000 ~ 90000 切一个 region，90000 ~ +inf 切一个 region ，可以通过手动指定点来切分 region，示例如下：
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE t BY (10000), (90000);
+```
+
+### Split Index region
+
+表中索引数据的 key 由 `table_id`、`index_id` 以及索引列的值编码组成，格式如下：
+
+```go
+t[table_id]_i[index_id][index_value]
+```
+
+例如，当 `table_id` 是 22，`index_id` 是 5，`index_value` 是 abc 时：
+
+```go
+t22_i5abc
+```
+
+同一表中同一索引数据的 `table_id` 和 `index_id` 是一样的，所以切分索引 region 要根据 `index_value` 来切分 region。
+
+#### 均匀切分
+
+索引均匀切分与行数据均匀切分的原理一样，只是计算 step 的值较为复杂，因为 `index_value` 可能不是整数。
+
+`upper` 和 `lower` 的值会先编码成 byte 数组，去掉 `lower` 和 `upper` byte 数组的最长公共前缀后，从 `lower` 和 `upper` 各取前 8 字节转成 uint64，再计算 `step = (upper - lower)/num`。计算出 step 后再将 step 编码成 byte 数组，添加到之前 `upper`和 `lower`的最长公共前缀后面组成一个 key 后去做切分。示例如下：
+
+如果索引 idx 的列也是整数类型，可以用如下 SQL 语句切分索引数据：
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE t INDEX idx BETWEEN (-9223372036854775808) AND (9223372036854775807) REGIONS 16;
+```
+
+该语句会把表 t 中 idx 索引数据 region 从 `minInt64` 到 `maxInt64` 之间均匀切割出 16 个 region。
+
+如果索引 idx1 的列是 varchar 类型，希望根据前缀字母来切分索引数据：
+
+{{< copyable "sql" >}}
+
+```sql
+SPLIT TABLE t INDEX idx1 BETWEEN ("a") AND ("z") REGIONS 26;
+```
+
+该语句会把表 t 中 idx1 索引数据的 region 从 a
+
+
```

Comment:
Why are there two blank lines here?

Reply:
I fixed this in another PR https://github.com/pingcap/docs-cn/pull/1487.